test/unittests/execute_floating_point_test.cpp

```diff
@@ -538,6 +539,130 @@ TYPED_TEST(execute_floating_point_types, add)
     EXPECT_THAT(exec(TypeParam{0x0.287p2}, TypeParam{0x1.FFp4}), Result(TypeParam{0x1.048Ep5}));
 }
 
+TYPED_TEST(execute_floating_point_types, sub)
+{
+    using FP = FP<TypeParam>;
+    using Limits = typename FP::Limits;
+
+    auto instance = instantiate(parse(this->get_binop_code(Instr::f32_sub)));
+    const auto exec = [&](auto arg1, auto arg2) { return execute(*instance, 0, {arg1, arg2}); };
+
+    // fsub(+-inf, +-inf) = nan:canonical
+    EXPECT_THAT(exec(Limits::infinity(), Limits::infinity()), CanonicalNaN(TypeParam{}));
+    EXPECT_THAT(exec(-Limits::infinity(), -Limits::infinity()), CanonicalNaN(TypeParam{}));
+
+    // fsub(+-inf, -+inf) = +-inf
+    EXPECT_THAT(exec(Limits::infinity(), -Limits::infinity()), Result(Limits::infinity()));
+    EXPECT_THAT(exec(-Limits::infinity(), Limits::infinity()), Result(-Limits::infinity()));
+
+    // fsub(+-0, +-0) = +0
+    EXPECT_THAT(exec(TypeParam{0.0}, TypeParam{0.0}), Result(TypeParam{0.0}));
+    EXPECT_THAT(exec(-TypeParam{0.0}, -TypeParam{0.0}), Result(TypeParam{0.0}));
+
+    // fsub(+-0, -+0) = +-0
+    EXPECT_THAT(exec(TypeParam{0.0}, -TypeParam{0.0}), Result(TypeParam{0.0}));
+    EXPECT_THAT(exec(-TypeParam{0.0}, TypeParam{0.0}), Result(-TypeParam{0.0}));
+
+    // fsub(z1, +-0) = z1  (for z1 = +-inf)
```

Comment:
I don't see this (+-inf, +-0) specified in the screenshot?

Reply:
You have literally fsub(z1, +-0) = z1 in the spec. Here is the variant of it with z1 being inf.

Reply:
Sorted the question out by writing out all formulas and checking it off one by one.